### PR TITLE
roachtest: bump jepsen version

### PR DESCRIPTION
**Pull request by renatolabs**
This commit bumps the jepsen version now to include a fix that allows
jepsen to set custom range sizes. See:

https://github.com/cockroachdb/jepsen/pull/34

Epic: none

Release note: None

### CODE REVIEW

**Review by smg260 — APPROVED** [review]
> Reviewed 1 of 1 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/99669)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @srosenberg)

**Comment by renatolabs** [comment]
Sadly this will not work as Jepsen invokes `sudo` to invoke cockroach, which will erase any inherited environment we may set here.

**Comment by renatolabs** [comment]
Jepsen test passes with these changes: <https://teamcity.cockroachdb.com/viewLog.html?buildId=9277206&buildTypeId=Cockroach_Nightlies_RoachtestNightlyGceBazel&tab=testsInfo&branch_Cockroach_Nightlies=99669>.

I'll wait for an approval in https://github.com/cockroachdb/jepsen/pull/34, merge that, generate a new uberjar, upload that to GCS, then update this PR to point to the new jar.

**Comment by renatolabs** [comment]
Seems like the change in cockroachdb/jepsen#34 was not the full story. Running jepsen tests on this branch leads to ~50% of Jepsen tests still failing: <https://teamcity.cockroachdb.com/viewLog.html?buildId=9301827&tab=buildResultsDiv&buildTypeId=Cockroach_Nightlies_RoachtestNightlyGceBazel#testNameId2565973610725110455>.

This will need further investigation.

**Comment by renatolabs** [comment]
All Jepsen tests pass now: <https://teamcity.cockroachdb.com/viewLog.html?buildId=9320590&buildTypeId=Cockroach_Nightlies_RoachtestNightlyGceBazel>.

@smg260 can I get a new stamp please?

**Review by smg260 — APPROVED** [review]
> Reviewed 1 of 1 files at r5, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/99669)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @srosenberg)

**Review by smg260 — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/99669)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @srosenberg)

**Comment by renatolabs** [comment]
bors r=smg260

**Pull request merged**
